### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by dariia-romanova**
[DEMO LINK](https://dariia-romanova.github.io/react_movies-list-fetch-movies/)

### CODE REVIEW

**Review by mariia-bila — APPROVED** [review]
Well done!

File: src/components/FindMovie/FindMovie.tsx [hunk 1]
```diff
@@ -1,54 +1,111 @@
...
+        setFoundStatus(false);
+      } else {
+        showMovie(movie);
+        setFoundStatus(true);
+      }
+    } catch (error) {
+      showMovie(null);
+      setFoundStatus(false);
+    }
+  };
```
Comment (mariia-bila) [inline src/components/FindMovie/FindMovie.tsx:34]: ` showMovie(null);
  setFoundStatus(false);`

no need to write this code twice

File: src/App.tsx [hunk 2]
```diff
@@ -12,6 +12,20 @@ export class App extends Component<{}, State> {
...
+    this.setState((state) => {
+      const isAdded = state.movies.some(({ imdbID }) => imdbID === movie.imdbID);
+
+      if (!isAdded) {
+        return {
+          movies: [...state.movies, movie],
+        };
+      }
+
+      return { movies: state.movies };
```
Comment (AlinaMatuschak) [inline src/App.tsx:25]:
```suggestion
      return state;
```

File: src/components/FindMovie/FindMovie.tsx [hunk 3]
```diff
@@ -1,54 +1,118 @@
...
-          />
-        </div>
+import { MovieCard } from '../MovieCard';
+
+type Props = {
+  addMovie: (movie: Movie) => void;
+};
+
+export const FindMovie: React.FC<Props> = (props) => {
+  const { addMovie } = props;
```
Comment (AlinaMatuschak) [inline src/components/FindMovie/FindMovie.tsx:13]:
```suggestion
export const FindMovie: React.FC<Props> = ({ addMovie }) => {
```

File: src/components/FindMovie/FindMovie.tsx [hunk 4]
```diff
@@ -1,54 +1,118 @@
...
+
+  const [title, setTitle] = useState('');
+  const [foundMovie, showMovie] = useState<Movie | null>(null);
+  const [foundStatus, setFoundStatus] = useState(true);
```
Comment (AlinaMatuschak) [inline src/components/FindMovie/FindMovie.tsx:17]:
```suggestion
  const [isFound, setIsFound] = useState(true);
```

**Review by AlinaMatuschak — APPROVED** [review]
Awesome 🤩

**Pull request closed without merging**